proposals/interface-types/working-notes/Instructions.md

```diff
@@ -0,0 +1,369 @@
+# Interface instructions
+
+This proposal defines new instructions that operate over interface value types. These instructions have the same form as core instructions; they push and pop values on an execution stack and support single pass validation.
+
+The new interface instructions are summarized below.
+
+```
+#interface-instr ::=
+  | bool.lift_iMM
+  | bool.lower_iMM
+
+  | sNN.lift_iMM
+  | sNN.lower_iMM
+
+  | uNN.lift_iMM
+  | uNN.lower_iMM
+
+  | string.lift_memory ..
+  | string.lower_memory ..
+
+  | variant.lift ..
+  | variant.lower_tag ..
+  | variant.lower ..
+
+  | record.lift ..
+  | record.lower ..
+
+  | array.lift_memory ..
+  | array.lower_memory ..
+
+  | callback.lift ..
```

Comment:
callbacks SHOULD involve their own import and export adapters

Reply:
I've temporarily dropped `callback` types from this PR as it seems that there are important details to resolve that I don't want to hang up the rest of this PR on.